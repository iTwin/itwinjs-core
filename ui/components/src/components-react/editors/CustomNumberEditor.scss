/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/core-react/lib/core-react/style/variables";
@import "./variables";

.components-customnumber-editor {
  width: 100%;
  height: 100%;
  margin: 0;
}

.components-smallEditor-host {
  .components-customnumber-editor {
    &.iui-input {
      min-height: $uicore-component-height-small !important;
      padding-top: 1px !important;
      padding-bottom: 1px !important;
    }
  }
}