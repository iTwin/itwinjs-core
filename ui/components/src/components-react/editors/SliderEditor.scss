/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/core-react/lib/core-react/style/variables";

.components-slider-editor {
  min-width: 100%;
}

.components-slider-editor-slider {
  min-width: 100px;
  min-height: $uicore-component-height-small;
}