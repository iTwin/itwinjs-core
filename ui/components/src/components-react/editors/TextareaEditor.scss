/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/core-react/lib/cjs/core-react/style/colors";

.components-textarea-editor {
  min-width: 100%;
  margin: 0;
}

.components-textarea-editor-textarea {
  min-width: 150px;
  margin: 0;
}