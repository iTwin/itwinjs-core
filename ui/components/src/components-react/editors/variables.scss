/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/core-react/lib/core-react/inputs/variables";

$editor-iui-input-padding: $uicore-iui-input-padding;