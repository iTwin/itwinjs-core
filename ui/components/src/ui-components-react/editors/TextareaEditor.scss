/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/ui-core-react/lib/ui-core-react/style/colors";

.components-textarea-editor {
  min-width: 100%;
  margin: 0;
}

.components-textarea-editor-textarea {
  min-width: 150px;
  margin: 0;
}