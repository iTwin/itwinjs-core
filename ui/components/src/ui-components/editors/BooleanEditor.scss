/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/itwinui-css/scss/style/space";
@import "~@itwin/itwinui-css/scss/style/typography";

$checkbox-size: $iui-font-size * 1.2;

.components-boolean-editor {
  height:      $checkbox-size;
  width:       $checkbox-size;
  margin-left: 0px;

  &.components-cell-editor {
    border:      none;
  }
}

.components-property-list--horizontal {
  .components-boolean-editor {
    margin-left:        $iui-s;    // Clear the resizer in the PropertyGrid
  }
}
