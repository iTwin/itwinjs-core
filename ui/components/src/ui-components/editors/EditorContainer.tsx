/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module PropertyEditors
 */

import "./EditorContainer.scss";
import * as React from "react";
import { IModelApp, NotifyMessageDetails } from "@bentley/imodeljs-frontend";
import { PropertyRecord, PropertyValue, SpecialKey } from "@bentley/ui-abstract";
import { CommonProps } from "@bentley/ui-core";
import { AsyncErrorMessage, PropertyEditorBase, PropertyEditorManager } from "./PropertyEditorManager";

/** Arguments for the Property Updated event callback
 * @beta
 */
export interface PropertyUpdatedArgs {
  /** The property being updated. */
  propertyRecord: PropertyRecord;
  /** The new value for the property. */
  newValue: PropertyValue;
}

/** Properties for a property editor component
 * @beta
 */
export interface PropertyEditorProps extends CommonProps {
  /** The property being updated. */
  propertyRecord?: PropertyRecord;
  /** Handler for commit */
  onCommit?: (args: PropertyUpdatedArgs) => void;
  /** Handler for cancel */
  onCancel?: () => void;
  /** Handler for blur */
  onBlur?: (event: React.FocusEvent) => void;
  /** Indicates whether the Property Editor should set focus */
  setFocus?: boolean;
}

/** [[EditorContainer]] React component properties
 * @beta
 */
export interface EditorContainerProps extends CommonProps {
  /** The property being updated. */
  propertyRecord: PropertyRecord;
  /** Tooltip text */
  title?: string;
  /** Handler for commit */
  onCommit: (args: PropertyUpdatedArgs) => void;
  /** Handler for cancel */
  onCancel: () => void;
  /** Indicates whether the Property Editor should set focus */
  setFocus?: boolean;

  /** @internal */
  ignoreEditorBlur?: boolean;
}

/** @internal */
interface CloneProps extends PropertyEditorProps {
  ref: (ref: any) => void;
}

/** Interface implemented by React based type editors
 * @beta
 */
export interface TypeEditor {
  getPropertyValue: () => Promise<PropertyValue | undefined>;
  htmlElement: HTMLElement | null;
  hasFocus: boolean;
}

/**
 * EditorContainer React component used by the Table, Tree and PropertyGrid for cell editing.
 * @beta
 */
export class EditorContainer extends React.PureComponent<EditorContainerProps> {

  private _editorRef: TypeEditor | undefined;
  private _propertyEditor: PropertyEditorBase | undefined;
  private _spanRef = React.createRef<HTMLSpanElement>();

  private createEditor(): React.ReactNode {
    const editorRef = (ref: TypeEditor | undefined) => this._editorRef = ref;

    const editorProps: CloneProps = {
      ref: editorRef,
      onCommit: this._handleEditorCommit,
      onCancel: this._handleEditorCancel,
      onBlur: this._handleEditorBlur,
      propertyRecord: this.props.propertyRecord,
      setFocus: this.props.setFocus !== undefined ? this.props.setFocus : true,
      className: this.props.className,
      style: this.props.style,
    };

    const propDescription = this.props.propertyRecord.property;

    const editorName = propDescription.editor !== undefined ? propDescription.editor.name : undefined;
    this._propertyEditor = PropertyEditorManager.createEditor(propDescription.typename, editorName, propDescription.dataController);
    const editorNode: React.ReactNode = this._propertyEditor.reactNode;

    let clonedNode: React.ReactNode = null;
    // istanbul ignore else
    if (React.isValidElement(editorNode)) {
      clonedNode = React.cloneElement(editorNode, editorProps);
    }

    return clonedNode;
  }
  public componentDidMount() {
    this._spanRef.current?.addEventListener("keydown", this._handleKeyDown, true);
  }
  public componentWillUnmount() {
    // istanbul ignore next
    if (this._spanRef.current)
      this._spanRef.current.removeEventListener("keydown", this._handleKeyDown, true);
  }

  private _handleEditorBlur = (_e: React.FocusEvent) => {
    // istanbul ignore else
    if (!this.props.ignoreEditorBlur && this._propertyEditor && this._propertyEditor.containerHandlesBlur)
      this._handleContainerCommit(); // eslint-disable-line @typescript-eslint/no-floating-promises
  };

  private _handleContainerBlur = (e: React.FocusEvent) => {
    e.stopPropagation();
  };

  private _handleClick = (e: React.MouseEvent) => {
    e.stopPropagation();
  };

  private _handleKeyDown = (e: KeyboardEvent) => {
    switch (e.key) {
      case SpecialKey.Escape:
        this.onPressEscape(e);
        break;
      case SpecialKey.Enter:
        this.onPressEnter(e);
        break;
      case SpecialKey.Tab:
        this.onPressTab(e);
        break;
      default:
        e.stopPropagation();
    }
  };

  private _handleRightClick = (e: React.MouseEvent) => {
    e.stopPropagation();
  };

  private onPressEscape(_e: KeyboardEvent): void {
    // istanbul ignore else
    if (this._propertyEditor && this._propertyEditor.containerHandlesEscape) {
      this._commitCancel();
    }
  }

  private onPressEnter(e: KeyboardEvent): void {
    // istanbul ignore else
    if (this._propertyEditor && this._propertyEditor.containerHandlesEnter) {
      // istanbul ignore else
      if (this._editorRef && this._editorRef.hasFocus)
        e.stopPropagation();
      this._handleContainerCommit(); // eslint-disable-line @typescript-eslint/no-floating-promises
    }
  }

  private onPressTab(e: KeyboardEvent): void {
    // istanbul ignore else
    if (this._propertyEditor && this._propertyEditor.containerHandlesTab) {
      e.stopPropagation();
      this._handleContainerCommit(); // eslint-disable-line @typescript-eslint/no-floating-promises
    }
  }

  private displayInputFieldMessage(errorMessage: AsyncErrorMessage | undefined) {
    // istanbul ignore else
    if (errorMessage && this._editorRef) {
      const details = new NotifyMessageDetails(errorMessage.priority, errorMessage.briefMessage, errorMessage.detailedMessage);
      const htmlElement = this._editorRef && this._editorRef.htmlElement;
      // istanbul ignore else
      if (htmlElement)
        details.setInputFieldTypeDetails(htmlElement);
      if (IModelApp.notifications)
        IModelApp.notifications.outputMessage(details);
    }
  }

  private async isNewValueValid(value: PropertyValue): Promise<boolean> {
    let isValid = true;

    // istanbul ignore else
    if (this._propertyEditor && this.props.propertyRecord) {
      const validateResult = await this._propertyEditor.validateValue(value, this.props.propertyRecord);

      if (validateResult.encounteredError) {
        this.displayInputFieldMessage(validateResult.errorMessage);
        isValid = false;
      }
    } else {
      isValid = false;
    }

    return isValid;
  }

  private _handleEditorCommit = (args: PropertyUpdatedArgs): void => {
    this._commit(args); // eslint-disable-line @typescript-eslint/no-floating-promises
  };

  private _handleContainerCommit = async (): Promise<void> => {
    const newValue = this._editorRef && await this._editorRef.getPropertyValue();
    // istanbul ignore else
    if (newValue !== undefined) {
      this._commit({ propertyRecord: this.props.propertyRecord, newValue });  // eslint-disable-line @typescript-eslint/no-floating-promises
    }
  };

  private _commit = async (args: PropertyUpdatedArgs) => {
    const newValue = args.newValue;
    const isValid = await this.isNewValueValid(newValue);
    if (isValid) {
      let doCommit = true;
      // istanbul ignore else
      if (this._propertyEditor && args.propertyRecord) {
        const commitResult = await this._propertyEditor.commitValue(newValue, args.propertyRecord);
        if (commitResult.encounteredError) {
          this.displayInputFieldMessage(commitResult.errorMessage);
          doCommit = false;
        }
      }

      if (doCommit) {
        this.props.onCommit(args);
      }
    }
  };

  private _handleEditorCancel = () => {
    this._commitCancel();
  };

  private _commitCancel = () => {
    this.props.onCancel();
  };

  /** @internal */
  public render() {
    return (
      <span className="components-editor-container"
        onBlur={this._handleContainerBlur}
        onClick={this._handleClick}
        onContextMenu={this._handleRightClick}
        title={this.props.title}
        data-testid="editor-container"
        role="presentation"
        ref={this._spanRef}
      >
        {this.createEditor()}
      </span>
    );
  }
}
