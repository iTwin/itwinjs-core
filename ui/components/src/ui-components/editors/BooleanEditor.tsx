/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module PropertyEditors
 */

import "./BooleanEditor.scss";
import classnames from "classnames";
import * as React from "react";
import { PropertyValue, PropertyValueFormat, StandardTypeNames } from "@bentley/ui-abstract";
import { Checkbox } from "@bentley/ui-core";
import { PropertyEditorProps, TypeEditor } from "./EditorContainer";
import { PropertyEditorBase, PropertyEditorManager } from "./PropertyEditorManager";

/** @internal */
interface BooleanEditorState {
  checkboxValue: boolean;
  isDisabled?: boolean;
}

/** BooleanEditor React component that is a property editor with checkbox input
 * @beta
 */
export class BooleanEditor extends React.PureComponent<PropertyEditorProps, BooleanEditorState> implements TypeEditor {
  private _isMounted = false;
  private _inputElement = React.createRef<HTMLInputElement>();

  /** @internal */
  public readonly state: Readonly<BooleanEditorState> = {
    checkboxValue: false,
    isDisabled: false,
  };

  public async getPropertyValue(): Promise<PropertyValue | undefined> {
    const record = this.props.propertyRecord;
    let propertyValue: PropertyValue | undefined;

    // istanbul ignore else
    if (record && record.value.valueFormat === PropertyValueFormat.Primitive) {
      propertyValue = {
        valueFormat: PropertyValueFormat.Primitive,
        value: this.state.checkboxValue,
        displayValue: "",
      };
    }

    return propertyValue;
  }

  public get htmlElement(): HTMLElement | null {
    return this._inputElement.current;
  }

  public get hasFocus(): boolean {
    return document.activeElement === this._inputElement.current;
  }

  private _updateCheckboxValue = (e: React.ChangeEvent<HTMLInputElement>) => {
    // istanbul ignore else
    if (this._isMounted) {
      let checkboxValue: boolean = false;

      // istanbul ignore if
      if (e.target.checked !== undefined)   // Needed for unit test environment
        checkboxValue = e.target.checked;
      else {
        // istanbul ignore else
        if (e.target.value !== undefined && typeof e.target.value === "boolean")
          checkboxValue = e.target.value;
      }

      this.setState({
        checkboxValue,
      }, async () => {
        // istanbul ignore else
        if (this.props.propertyRecord && this.props.onCommit) {
          const propertyValue = await this.getPropertyValue();
          // istanbul ignore else
          if (propertyValue !== undefined) {
            this.props.onCommit({ propertyRecord: this.props.propertyRecord, newValue: propertyValue });
          }
        }
      });
    }
  };

  /** @internal */
  public componentDidMount() {
    this._isMounted = true;
    this.setStateFromProps(); // eslint-disable-line @typescript-eslint/no-floating-promises
  }

  /** @internal */
  public componentWillUnmount() {
    this._isMounted = false;
  }

  /** @internal */
  public componentDidUpdate(prevProps: PropertyEditorProps) {
    if (this.props.propertyRecord !== prevProps.propertyRecord) {
      this.setStateFromProps(); // eslint-disable-line @typescript-eslint/no-floating-promises
    }
  }

  private async setStateFromProps() {
    const { propertyRecord } = this.props;
    let checkboxValue = false;
    let isDisabled = false;

    // istanbul ignore else
    if (propertyRecord && propertyRecord.value.valueFormat === PropertyValueFormat.Primitive) {
      const primitiveValue = propertyRecord.value.value;
      checkboxValue = primitiveValue as boolean;
    }

    // istanbul ignore else
    if (propertyRecord && propertyRecord.isDisabled)
      isDisabled = propertyRecord.isDisabled;

    // istanbul ignore else
    if (this._isMounted)
      this.setState({ checkboxValue, isDisabled });
  }

  /** @internal */
  public render() {
    const className = classnames("components-cell-editor", "components-boolean-editor", this.props.className);
    const checked = this.state.checkboxValue;
    const isDisabled = !!this.state.isDisabled;

    return (
      <Checkbox
        inputRef={this._inputElement}
        onBlur={this.props.onBlur}
        className={className}
        style={this.props.style}
        checked={checked}
        onChange={this._updateCheckboxValue}
        setFocus={this.props.setFocus}
        disabled={isDisabled}
        data-testid="components-checkbox-editor">
      </Checkbox>
    );
  }
}

/** Boolean Property Editor registered for the "bool" and "boolean" type names.
 * It uses the [[BooleanEditor]] React component.
 * @beta
 */
export class BooleanPropertyEditor extends PropertyEditorBase {
  public get reactNode(): React.ReactNode {
    return <BooleanEditor />;
  }
}

PropertyEditorManager.registerEditor(StandardTypeNames.Bool, BooleanPropertyEditor);
PropertyEditorManager.registerEditor(StandardTypeNames.Boolean, BooleanPropertyEditor);
