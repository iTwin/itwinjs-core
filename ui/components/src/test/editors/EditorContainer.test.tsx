/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/

import { expect } from "chai";
import { mount, shallow } from "enzyme";
import * as React from "react";
import sinon from "sinon";
import { EditorContainer, PropertyUpdatedArgs } from "../../ui-components/editors/EditorContainer";
import TestUtils from "../TestUtils";
import { SpecialKey, StandardEditorNames } from "@bentley/ui-abstract";
import { cleanup, fireEvent, render } from "@testing-library/react";

describe("<EditorContainer />", () => {
  before(async () => {
    await TestUtils.initializeUiComponents();
  });

  beforeEach(() => {
    sinon.restore();
  });

  afterEach(cleanup);

  after(() => {
    TestUtils.terminateUiComponents();
  });

  it("should render", () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    const sut = mount(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={() => { }} onCancel={() => { }} />);
    sut.unmount();
  });

  it("renders correctly", () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    shallow(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={() => { }} onCancel={() => { }} />).should.matchSnapshot();
  });

  it("renders editor for 'text' type using TextEditor", () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    const wrapper = mount(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={() => { }} onCancel={() => { }} />);
    expect(wrapper.find("input.components-text-editor").length).to.eq(1);
    wrapper.unmount();
  });

  it("calls onCommit for Enter", async () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    const spyOnCommit = sinon.spy();
    function handleCommit(_commit: PropertyUpdatedArgs): void {
      spyOnCommit();
    }
    const wrapper = render(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={handleCommit} onCancel={() => { }} />);
    const inputNode = wrapper.container.querySelector("input");
    expect(inputNode).not.to.be.null;

    fireEvent.keyDown(inputNode as HTMLElement, { key: SpecialKey.Enter })
    await TestUtils.flushAsyncOperations();
    expect(spyOnCommit.calledOnce).to.be.true;
  });

  it("calls onCancel for Escape", async () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    const spyOnCancel = sinon.spy();
    const wrapper = render(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={() => { }} onCancel={spyOnCancel} />);
    const inputNode = wrapper.container.querySelector("input");
    expect(inputNode).not.to.be.null;

    fireEvent.keyDown(inputNode as HTMLElement, { key: SpecialKey.Escape })
    await TestUtils.flushAsyncOperations();
    expect(spyOnCancel.calledOnce).to.be.true;
  });

  it("calls onCancel for Cancel button in popup", async () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value", undefined, { name: StandardEditorNames.MultiLine });
    const spyOnCancel = sinon.spy();
    const wrapper = mount(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={() => { }} onCancel={spyOnCancel} />);

    const button = wrapper.find(".components-popup-button");
    expect(button.length).to.eq(1);
    button.first().simulate("click");
    await TestUtils.flushAsyncOperations();

    const okButton = wrapper.find("button.components-popup-cancel-button");
    expect(okButton.length).to.eq(1);
    okButton.first().simulate("click");
    await TestUtils.flushAsyncOperations();

    expect(spyOnCancel.calledOnce).to.be.true;
    wrapper.unmount();
  });

  it("calls onCommit for Tab", async () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    const spyOnCommit = sinon.spy();
    function handleCommit(_commit: PropertyUpdatedArgs): void {
      spyOnCommit();
    }
    const wrapper = render(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={handleCommit} onCancel={() => { }} />);
    const inputNode = wrapper.container.querySelector("input");
    expect(inputNode).not.to.be.null;

    fireEvent.keyDown(inputNode as HTMLElement, { key: SpecialKey.Tab })
    await TestUtils.flushAsyncOperations();
    expect(spyOnCommit.calledOnce).to.be.true;
  });

  it("processes other input node events", () => {
    const propertyRecord = TestUtils.createPrimitiveStringProperty("Test1", "my value");
    const spyOnCommit = sinon.spy();
    function handleCommit(_commit: PropertyUpdatedArgs): void {
      spyOnCommit();
    }
    const wrapper = mount(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={handleCommit} onCancel={() => { }} />);
    const inputNode = wrapper.find("input");
    expect(inputNode.length).to.eq(1);

    inputNode.simulate("blur");
    inputNode.simulate("click");
    inputNode.simulate("contextMenu");

    const renderedWrapper = render(<EditorContainer propertyRecord={propertyRecord} title="abc" onCommit={handleCommit} onCancel={() => { }} />);
    const renderedInputNode = renderedWrapper.container.querySelector("input");
    fireEvent.keyDown(renderedInputNode as HTMLElement, { key: SpecialKey.ArrowLeft });
    wrapper.unmount();
  });

});
