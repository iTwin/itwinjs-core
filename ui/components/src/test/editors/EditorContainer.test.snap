// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditorContainer /> renders correctly 1`] = `
<span
  className="components-editor-container"
  data-testid="editor-container"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  role="presentation"
  title="abc"
>
  <TextEditor
    onBlur={[Function]}
    onCancel={[Function]}
    onCommit={[Function]}
    propertyRecord={
      PropertyRecord {
        "isReadonly": false,
        "property": Object {
          "displayLabel": "Test1",
          "name": "Test1",
          "typename": "string",
        },
        "value": Object {
          "displayValue": "my value",
          "value": "my value",
          "valueFormat": 0,
        },
      }
    }
    setFocus={true}
  />
</span>
`;
